firebase-firestore/src/main/java/com/google/firebase/firestore/AggregateQuery.java

```diff
@@ -0,0 +1,88 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore;
+
+import androidx.annotation.NonNull;
+import com.google.android.gms.tasks.Task;
+import com.google.android.gms.tasks.TaskCompletionSource;
+import com.google.firebase.firestore.util.Executors;
+
+/**
+ * A {@code AggregateQuery} computes some aggregation statistics from the result set of a base
+ * {@link Query}.
+ */
+public final class AggregateQuery {
+
+  // The base query.
+  private final Query query;
+
+  AggregateQuery(@NonNull Query query, @NonNull AggregateField aggregateField) {
+    this.query = query;
+    if (!(aggregateField instanceof AggregateField.CountAggregateField)) {
+      throw new IllegalArgumentException("unsupported aggregateField: " + aggregateField);
+    }
+  }
+
+  /** Returns the base {@link Query} for this aggregate query. */
+  @NonNull
+  public Query getQuery() {
+    return query;
+  }
+
+  /**
+   * Executes the aggregate query and returns the results as a {@code AggregateQuerySnapshot}.
+   *
+   * @param source A value to configure the get behavior.
+   * @return A Task that will be resolved with the results of the {@code AggregateQuery}.
+   */
+  @NonNull
+  public Task<AggregateQuerySnapshot> get(@NonNull AggregateSource source) {
```

Comment:
Consider verifying that `source` is not null using `Preconditions.checkNotNull()`. Also, add a unit test for specifying null.

Reply:
Done.